packages/apps/job-launcher/server/src/modules/auth/strategy/jwt.http.ts

```diff
@@ -6,57 +6,40 @@ import { ConfigService } from '@nestjs/config';
 import { UserEntity } from '../../user/user.entity';
 import { UserStatus } from '../../../common/enums/user';
 import { ConfigNames } from '../../../common/config';
-import { AuthService } from '../auth.service';
-import { JWT_PREFIX } from '../../../common/constants';
-import { AuthRepository } from '../auth.repository';
+import { UserRepository } from 'src/modules/user/user.repository';
 
 @Injectable()
 export class JwtHttpStrategy extends PassportStrategy(Strategy, 'jwt-http') {
   constructor(
-    private readonly authRepository: AuthRepository,
-    private readonly authService: AuthService,
+    private readonly userRepository: UserRepository,
     private readonly configService: ConfigService,
   ) {
     super({
       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
       ignoreExpiration: false,
-      secretOrKey: configService.get<string>(
-        ConfigNames.JWT_SECRET,
-        'secretkey',
-      ),
+      secretOrKey: configService.get<string>(ConfigNames.JWT_PUBLIC_KEY, ''),
+      algorithm: ['RS256'],
       passReqToCallback: true,
     });
   }
 
   public async validate(
     @Req() request: any,
-    payload: { email: string; userId: number },
+    payload: { userId: number },
   ): Promise<UserEntity> {
-    const auth = await this.authRepository.findOneByUserId(payload.userId);
+    const user = await this.userRepository.findById(payload.userId);
 
-    if (!auth?.user) {
+    if (!user) {
       throw new UnauthorizedException('User not found');
     }
 
-    if (auth?.user.status !== UserStatus.ACTIVE) {
+    if (
+      user.status !== UserStatus.ACTIVE &&
+      request.url !== '/auth/resend-email-verification'
```

Comment:
Actually it will prevent to delete the user if tokens exist.
To avoid accidental data loss, I prefer to get an error raised saying that it cannot be deleted because tokens data exist, so I manually delete tokens first, then the user if this what actually wanted to do.